presto-main/src/main/java/com/facebook/presto/operator/PartitionedOutputOperator.java

```diff
@@ -317,6 +332,7 @@ public PagePartitioner(
             this.partitionConstants = requireNonNull(partitionConstants, "partitionConstants is null").stream()
                     .map(constant -> constant.map(NullableValue::asBlock))
                     .collect(toImmutableList());
+            this.replicateAny = requireNonNull(replication, "replication is null") == Replication.REPLICATE_NULLS_AND_ANY;
```

Comment:
please add `replicateAny()` method to `Replication` enum (wrap enum logic into methods). Then you also wouldn't need `replicateAny` field here